src/oscar/apps/catalogue/managers.py

```diff
@@ -17,16 +17,25 @@ def base_queryset(self):
     def browsable(self):
         """
         Excludes non-canonical products.
+
+        Method is deprecated. Use BrowsableProductQuerySet instead.
         """
         return self.filter(parent=None)
 
+class BrowsableProductQuerySet(ProductQuerySet):
+
+    def base_queryset(self):
+        """
+        Excludes non-canonical products
+        """
+        return super().base_queryset().filter(parent=None)
+
 
 class ProductManager(models.Manager):
```

Comment:
Please apply the `oscar.core.decorators.deprecated` decorator to the deprecated classes - this raises a warning whenever they are instantiated.